src/classes/BDI_DataImport_BATCH.cls

```diff
@@ -465,8 +504,15 @@ public with sharing class BDI_DataImport_BATCH implements Database.Batchable<sOb
                 setEmail.add(di.Contact1_Work_Email__c);            
             if (di.Contact1_Alternate_Email__c != null)
                 setEmail.add(di.Contact1_Alternate_Email__c);            
-            if (isCustomIDInContactMatchRules && di.get('Contact1_' + diSettings.Contact_Custom_Unique_ID__c + strTestContactUniqueIdSuffix) != null)
-                setCustomID.add(string.valueOf(di.get('Contact1_' + diSettings.Contact_Custom_Unique_ID__c + strTestContactUniqueIdSuffix)));
+            if (isCustomIDInContactMatchRules) { 
+                object val = di.get('Contact1_' + diSettings.Contact_Custom_Unique_ID__c + strTestContactUniqueIdSuffix);
```

Comment:
It's a hack.  In the non-test scenario, the settings for the Account and Contact Custom Unique ID field is always a custom field, and the setting would contain the developer name of the field (and thus have __c).  But for my tests to try this functionality, I can't create any custom fields, and thus have to use some Standard Salesforce fields on either Contact or Account.  Thus I have to put the Standard field name in the setting, and use that when copying to the Contact or Account.  But that standard field which is also on the BDI object has the __c since the BDI is a custom object.  So it's a hack to say, if we are running tests, and the setting to use a custom id field is set, then we know it's a standard field against contact or account, but we still need to add the __c to access the field in the BDI object.